evm/vm/forks/frontier/blocks.py

```diff
@@ -243,6 +244,10 @@ def from_header(cls, header, db):
                 sedes=CountableList(BlockHeader),
             )
 
+        # XXX: This doesn't need to be obtained from chaindb because we have no need to override
```

Comment:
A brief perusal of the code suggest we could do the following.

* Alter the `VM` class to expect a `ChainDB` instance as it's `db` argument.
* Modify the various block and transaction lookups that happen internally in the `VM` class to use `ChainDB` methods.
* Move the `state_db` API onto the `ChainDB` class.

If I'm reading the code right, this would remove all direct db access by the `VM` class which seems good.  Thoughts?  Did I completely miss what the problem was?

Reply:
I think that should work, but then we'd also need to either extend ChainDB to implement the BaseDB API, or pass ChainDB.db to JournalDB

Reply:
Actually, to move the state_db API onto ChainDB, we'd need to move VM.journal_db there as well, which may be a good thing as then we could expose the snapshot/revert/commit APIs via ChainDB, which would allow us to get rid of VM.journal_db